qa/sc_evm_forger_stakes_pager.py

```diff
@@ -0,0 +1,357 @@
+    end_offset = 2 * 32
+    next_pos = decode(['int32'], hex_str_to_bytes(abi_return_value[start_offset:end_offset]))[0]
+
+    # next we have the offset at which dynamic data start, should be 64 (0x40)
+    start_offset = end_offset
+    end_offset = end_offset + 2 * 32
+    dynamic_data_start_offset = decode(['uint32'], hex_str_to_bytes(abi_return_value[start_offset:end_offset]))[0]
+    assert_equal(dynamic_data_start_offset, 64)
+
+    # size of the dynamic array of stakes
+    start_offset = end_offset
+    end_offset = end_offset + 2 * 32  # read 32 bytes
+    list_size = decode(['uint32'], hex_str_to_bytes(abi_return_value[start_offset:end_offset]))[0]
+
+    list_result = []
+    for i in range(list_size):
+        entry_dict = {}
+
+        start_offset = end_offset
+        end_offset = end_offset + 2 * 6 * 32  # read 6 32-bytes-long chunks of data
+        (stake_id_bytes,
+         value,
+         address,
+         block_sign_pub_key_bytes,
+         vrf32_bytes,
+         vrf1_bytes
+         ) = decode(['bytes32', 'uint256', 'address', 'bytes32', 'bytes32', 'bytes1'],
+                    hex_str_to_bytes(abi_return_value[start_offset:end_offset]))
+
+        stake_id = hexlify(stake_id_bytes).decode('ascii')
+        block_sign_pub_key = hexlify(block_sign_pub_key_bytes).decode('ascii')
+        vrf_key = hexlify(vrf32_bytes + vrf1_bytes).decode('ascii')
+
+        # fill an entry exactly as HTTP API would do
+        entry_dict['stakeId'] = stake_id
+        entry_dict['forgerStakeData'] = {
+            'forgerPublicKeys': {
+                'blockSignPublicKey': {'publicKey': block_sign_pub_key},
+                'vrfPublicKey': {'publicKey': vrf_key}},
+            'ownerPublicKey': {'address': remove_0x_prefix(address)},
+            'stakedAmount': value
+        }
+
+        list_result.append(entry_dict)
+
+    return next_pos, list_result
+
+
+def padded_32_hex(s):
+    return remove_0x_prefix(s).zfill(64)
+
+
+def get_paged_forging_stakes_via_eth_call(sc_node, from_address, start_pos, page_size):
+    # execute native smart contract for getting all associations
+    method = 'getPagedForgersStakes(uint32,uint32)'
+    abi_str = function_signature_to_4byte_selector(method)
+    start_pos = padded_32_hex(hex(start_pos))
+    size_padded_str = padded_32_hex(hex(page_size))
+
+    req = {
+        "from": format_evm(from_address),
+        "to": format_evm(FORGER_STAKE_SMART_CONTRACT_ADDRESS),
+        "gasLimit": 2300000,
+        "gasPrice": 950000000,
+        "value": 0,
+        "data": encode_hex(abi_str) + start_pos + size_padded_str
+    }
+    response = sc_node.rpc_eth_call(req, 'latest')
+    if 'error' in response:
+        raise RuntimeError("Something went wrong, see {}".format(str(response)))
+
+    abi_return_value = remove_0x_prefix(response['result'])
+    # print(abi_return_value)
+    return get_paged_stakes_from_abi(abi_return_value)
+
+
+class SCEvmForgerStakesPager(AccountChainSetup):
```

Comment:
This test misses the --allforks option support

Reply:
added a check at the beginning  of the test.